### PR TITLE
Fix template geenration to details

### PR DESCRIPTION
**Pull request by btwseeu78**
/kind bug
/kind cleanup
/kind feature

#### What this PR does / why we need it:
1. Current helm chart has no way to configure host for ingress in kueueviz it ends up referring  a random TLS-Secret(bug)
2. PriorityClass only added to manager not extended to kueueviz ,  frontend or backend deployments

Fixes #
-  https://github.com/kubernetes-sigs/kueue/issues/5437

## Previously Closed MR(for info):
https://github.com/kubernetes-sigs/kueue/pull/5754

#### Does this PR introduce a user-facing change?
```release-note
1. Add Support for configuring host for ingress in kueueviz.
2. changing PriorityClassName support extended to kueueviz  frontend or backend deployments.
3. Add imagePullSecrets for kueueViz frontend and backend deployments
4. Added test cases for kueueviz components in helm unit-test 
```

### CODE REVIEW

File: hack/processing-plan.yaml [hunk 1]
```diff
@@ -580,6 +580,19 @@ files:
       - kustomization.yaml
     continueOnError: true
     operations:
+      - type: DELETE
```
Comment (mbobrovskyi) [inline hack/processing-plan.yaml]: Shouldn't we just replace the environment variable value instead? Using `type: UPDATE`.

File: hack/processing-plan.yaml [hunk 2]
```diff
@@ -580,6 +580,19 @@ files:
...
+        key: .spec.template.spec.containers[0].env
+      - type: DELETE
```
Comment (mbobrovskyi) [inline hack/processing-plan.yaml]: The same question here. It looks like much easier just update it instead of delete and add on postOperations.

File: hack/processing-plan.yaml (hunk 1 above)
Comment (btwseeu78) [reply hack/processing-plan.yaml]:
```yaml
value: '"wss://{{ .Values.kueueViz.backend.ingress.host | default \"backend.kueueviz.local\" }}"'
```
i tried with this but it did not worked with the default.the templating engine does not taking the strings correctly.

the result is : 
```yaml
      env:
        - name: REACT_APP_WEBSOCKET_URL
          value: "{{ .Values.kueueViz.backend.ingress.host | default \"backend.kueueviz.local\" }}"
  ```
which causing issue in helm because of this escape sequence.

Comment (btwseeu78) [reply hack/processing-plan.yaml]: @mbobrovskyi i am open to suggestion if you can suggest me for one block, i will update the rest

**Comment by mimowo** [comment]
@btwseeu78  could you please address the remaining comments.

This is not really a bug, but a small feature. I'm ok to cherry-pick small helm features like this.
/remove-kind bug

**Comment by btwseeu78** [comment]
@mimowo i am waiting for feedback from @mbobrovskyi , both of the comments belong to same topic on generating helm from the processing plan.

**Comment by mimowo** [comment]
ack @mbobrovskyi ptal

File: hack/processing-plan.yaml (hunk 1 above)
Comment (mbobrovskyi) [reply hack/processing-plan.yaml]:
```yaml
      - type: UPDATE
        key: .spec.template.spec.containers.[].env.[].value
        value: '"wss://{{ .Values.kueueViz.backend.ingress.host | default \"backend.kueueviz.local\" }}"'
        onFileCondition: '.kind == "Deployment" and .metadata.name | contains("kueueviz-frontend")'
        onItemCondition: '.spec.template.spec.containers.[].env.[].name == "REACT_APP_WEBSOCKET_URL"'
```

Comment (mbobrovskyi) [reply hack/processing-plan.yaml]: I think I figured out the problem. Try changing the quotes in `config/components/kueueviz/frontend-deployment.yaml`:

https://github.com/kubernetes-sigs/kueue/blob/63a7f6432040a3ee99a4b97aa00a217af9640840/config/components/kueueviz/frontend-deployment.yaml#L25

It looks like the YAML processor preserves the same quote style as the original.

File: hack/processing-plan.yaml (hunk 2 above)
Comment (mbobrovskyi) [reply hack/processing-plan.yaml]: Please also remove redundant: 

https://github.com/kubernetes-sigs/kueue/blob/63a7f6432040a3ee99a4b97aa00a217af9640840/hack/processing-plan.yaml#L619-L622

Comment (mbobrovskyi) [reply hack/processing-plan.yaml]:
```yaml
      - type: UPDATE
        key: .spec.tls.[].secretName
        value: '"{{ .Values.kueueViz.backend.ingress.tlsSecretName | default \"kueueviz-backend-tls\" }}"'
        onFileCondition: '.kind == "Ingress" and .metadata.name | contains("kueueviz-backend-ingress")'
        onItemCondition: '.spec.tls.[].secretName == "kueueviz-tls-secret"'
```

Comment (mbobrovskyi) [reply hack/processing-plan.yaml]:
```yaml
      - type: UPDATE
        key: .spec.tls.[].hosts
        value: '["{{ .Values.kueueViz.backend.ingress.host | default \"backend.kueueviz.local\" }}"]'
        onFileCondition: '.kind == "Ingress" and .metadata.name | contains("kueueviz-backend-ingress")'
        onItemCondition: '.spec.tls.[].hosts.[] == "backend.kueueviz.local"'
```

File: hack/processing-plan.yaml (hunk 1 above)
Comment (btwseeu78) [reply hack/processing-plan.yaml]: > ```yaml
>       - type: UPDATE
>         key: .spec.template.spec.containers.[].env.[].value
>         value: '"wss://{{ .Values.kueueViz.backend.ingress.host | default \"backend.kueueviz.local\" }}"'
>         onFileCondition: '.kind == "Deployment" and .metadata.name | contains("kueueviz-frontend")'
>         onItemCondition: '.spec.template.spec.containers.[].env.[].name == "REACT_APP_WEBSOCKET_URL"'
> ```

this will be same as i had with:

```yaml
          env:
            - name: REACT_APP_WEBSOCKET_URL
              value: "wss://{{ .Values.kueueViz.backend.ingress.host | default \"backend.kueueviz.local\" }}"
```

Comment (btwseeu78) [reply hack/processing-plan.yaml]: Other Update patch is working as expected
so i will add them

Comment (mbobrovskyi) [reply hack/processing-plan.yaml]: Try to change quotes on the config/components/kueueviz/frontend-deployment.yaml.

Comment (btwseeu78) [reply hack/processing-plan.yaml]: Done! working

File: Makefile [hunk 3]
```diff
@@ -180,6 +180,10 @@ helm-verify: helm helm-lint ## run helm template and detect any rendering failur
 	$(HELM) template charts/kueue --set managerConfig.controllerManagerConfigYaml="managedJobsNamespaceSelector:\n  matchExpressions:\n    - key: kubernetes.io/metadata.name\n      operator: In\n      values: [ kube-system ]" > /dev/null
 # test priorityClassName option
 	$(HELM) template charts/kueue --set controllerManager.manager.priorityClassName="system-cluster-critical" > /dev/null
+# test priorityClassName option
+	$(HELM) template charts/kueue --set controllerManager.manager.priorityClassName="system-cluster-critical" > /dev/null
```
Comment (mbobrovskyi) [inline Makefile]: Is it duplication?

File: Makefile [hunk 4]
```diff
@@ -180,6 +180,10 @@ helm-verify: helm helm-lint ## run helm template and detect any rendering failur
...
+# test kueueViz priorityClassName options
+	$(HELM) template charts/kueue --set enableKueueViz=true --set kueueViz.priorityClassName="system-cluster-critical" > /dev/null
```
Comment (mbobrovskyi) [inline Makefile]: Can we also check other variables?

File: charts/kueue/templates/kueueviz/frontend-deployment.yaml [hunk 5]
```diff
@@ -22,7 +29,7 @@ spec:
             - containerPort: 8080
           env:
             - name: REACT_APP_WEBSOCKET_URL
-              value: "wss://backend.kueueviz.local"
+              value: wss://{{ .Values.kueueViz.backend.ingress.host | default "backend.kueueviz.local" }}
```
Comment (mbobrovskyi) [inline charts/kueue/templates/kueueviz/frontend-deployment.yaml]:
```suggestion
              value: 'wss://{{ .Values.kueueViz.backend.ingress.host | default "backend.kueueviz.local" }}'
```

Do we need single quotes like we have in other places?

File: charts/kueue/templates/kueueviz/backend-ingress.yaml [hunk 6]
```diff
@@ -8,12 +8,15 @@ metadata:
...
 spec:
+  {{- if .Values.kueueViz.backend.ingress.ingressClassName }}
+  ingressClassName: {{ .Values.kueueViz.backend.ingress.ingressClassName }}
+  {{- end }}
   tls:
     - hosts:
-        - backend.kueueviz.local
-      secretName: '{{ include "kueue.fullname" . }}-kueueviz-tls-secret'
+        - '{{ .Values.kueueViz.backend.ingress.host | default "backend.kueueviz.local" }}'
+      secretName: '{{ .Values.kueueViz.backend.ingress.tlsSecretName | default "kueueviz-backend-tls" }}'
```
Comment (mbobrovskyi) [inline charts/kueue/templates/kueueviz/backend-ingress.yaml:17]: Should we add `kueue.fullname` prefix for default? Like we did it before?

File: charts/kueue/templates/kueueviz/frontend-ingress.yaml [hunk 7]
```diff
@@ -8,12 +8,15 @@ metadata:
...
 spec:
+  {{- if .Values.kueueViz.frontend.ingress.ingressClassName }}
+  ingressClassName: {{ .Values.kueueViz.frontend.ingress.ingressClassName }}
+  {{- end }}
   tls:
     - hosts:
-        - frontend.kueueviz.local
-      secretName: '{{ include "kueue.fullname" . }}-kueueviz-tls-secret'
+        - '{{ .Values.kueueViz.frontend.ingress.host | default "frontend.kueueviz.local" }}'
+      secretName: '{{ .Values.kueueViz.frontend.ingress.tlsSecretName | default "kueueviz-frontend-tls" }}'
```
Comment (mbobrovskyi) [inline charts/kueue/templates/kueueviz/frontend-ingress.yaml:17]: And here.

File: Makefile (hunk 4 above)
Comment (btwseeu78) [reply Makefile]: yes i need to fix this line, and did not added lot of lines on verify since we can just test the edge conditions we dont use , i have added test cases in unittest for all features .

Comment (btwseeu78) [reply Makefile]: i removed the redundency

File: charts/kueue/templates/kueueviz/backend-ingress.yaml (hunk 6 above)
Comment (btwseeu78) [reply charts/kueue/templates/kueueviz/backend-ingress.yaml:17]: i don't think its matters much here, the only need is frontend REACT_ENV should be equal to backend host

Comment (mbobrovskyi) [reply charts/kueue/templates/kueueviz/backend-ingress.yaml:17]: Actually, maybe it's okay because we also need to add this prefix in values.yaml:

https://github.com/kubernetes-sigs/kueue/blob/89b4e6a1d405cfc5036be83542ad6439c37d0f67/charts/kueue/values.yaml#L220

File: charts/kueue/templates/kueueviz/frontend-deployment.yaml (hunk 5 above)
Comment (btwseeu78) [reply charts/kueue/templates/kueueviz/frontend-deployment.yaml]: chnaged the details

File: charts/kueue/templates/kueueviz/frontend-ingress.yaml (hunk 7 above)
Comment (btwseeu78) [reply charts/kueue/templates/kueueviz/frontend-ingress.yaml:17]: i think default being a constant value is fine.it does not impact core functionality nor connectivity.also without any host it will notwork for most use cases in general apart from serving the support for wss

File: charts/kueue/tests/kueue_test.yaml [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+    asserts:
+      - equal:
+          path: spec.template.spec.containers[0].env[0].value
+          value: "wss://backend.kueueviz.local"
+  - it: Render  Ingress for Backend Component
+    template: kueueviz/backend-ingress.yaml
+    set:
+      enableKueueViz: true
+      kueueViz:
+
```
Comment (mbobrovskyi) [inline charts/kueue/tests/kueue_test.yaml]:
```suggestion
```
nit

File: charts/kueue/tests/kueue_test.yaml [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+  - it: should not render the pod priorityClassName if not set for frontend deployment
+    template: kueueviz/frontend-deployment.yaml
+    set:
+      enableKueueViz: true
+      kueueViz:
+        priorityClassName:       
+    asserts:
+      - notExists:
+          path: spec.template.spec.priorityClassName
+  - it: should  render the ingress value of backend ingress in frontend deployment as ENV variables
```
Comment (mbobrovskyi) [inline charts/kueue/tests/kueue_test.yaml]:
```suggestion
  - it: should render the ingress value of backend ingress in frontend deployment as ENV variables
```
nit

File: charts/kueue/tests/kueue_test.yaml [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+        backend:
+          ingress:
+            ingressClassName: "ingress-external"
+            host: "backend.kueueviz.local"
+            tlsSecretName: "kueueviz-frontend-tls"
+    asserts:
+      - equal:
+          path: spec.template.spec.containers[0].env[0].value
+          value: "wss://backend.kueueviz.local"
+  - it: Render  Ingress for Backend Component
```
Comment (mbobrovskyi) [inline charts/kueue/tests/kueue_test.yaml]:
```suggestion
  - it: should render Ingress for backend component
```
nit

File: charts/kueue/tests/kueue_test.yaml [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+          path: spec.ingressClassName
+          value: "ingress-internal"
+      - equal:
+          path: spec.rules[0].host
+          value: "backend.kueueviz.in"
+      - equal:
+          path: spec.tls[0].secretName
+          value: "kueueviz-backend-tls"
+
+  - it: Render  Ingress for Frontend Component
```
Comment (mbobrovskyi) [inline charts/kueue/tests/kueue_test.yaml]:
```suggestion
  - it: should render Ingress for frontend component
```

File: charts/kueue/tests/kueue_test.yaml [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+    template: kueueviz/backend-ingress.yaml
+    set:
+      enableKueueViz: true
+      kueueViz:
+        
+        backend:
+          ingress:
+            ingressClassName: "ingress-internal"
+            host: "backend.kueueviz.in"
+            tlsSecretName: "kueueviz-backend-tls"
```
Comment (mbobrovskyi) [inline charts/kueue/tests/kueue_test.yaml]: Please avoid using the default value.

File: charts/kueue/tests/kueue_test.yaml [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+    template: kueueviz/frontend-ingress.yaml
+    set:
+      enableKueueViz: true
+      kueueViz:
+        ingressClassName: "ingress-external"
+        frontend:
+          ingress:
+            ingressClassName: "ingress-external"
+            host: "frontend.kueueviz.in"
+            tlsSecretName: "kueueviz-frontend-tls"
```
Comment (mbobrovskyi) [inline charts/kueue/tests/kueue_test.yaml]: And here.

File: charts/kueue/tests/kueue_test.yaml [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+      - equal:
+          path: spec.tls[0].secretName
+          value: "kueueviz-backend-tls"
+
+  - it: Render  Ingress for Frontend Component
+    template: kueueviz/frontend-ingress.yaml
+    set:
+      enableKueueViz: true
+      kueueViz:
+        ingressClassName: "ingress-external"
```
Comment (mbobrovskyi) [inline charts/kueue/tests/kueue_test.yaml]:
```suggestion
```

File: charts/kueue/tests/kueue_test.yaml (hunk 12 above)
Comment (btwseeu78) [reply charts/kueue/tests/kueue_test.yaml]: changes are done

File: charts/kueue/tests/kueue_test.yaml [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+    template: kueueviz/frontend-deployment.yaml
+    set:
+      enableKueueViz: true
+      kueueViz:
+        imagePullSecrets:
+          - name: "my-secret"
+    asserts:
+      - equal:
+          path: spec.template.spec.imagePullSecrets[0].name
+          value: "my-secret"
```
Comment (mbobrovskyi) [inline charts/kueue/tests/kueue_test.yaml:129]:
```suggestion
          value: "my-secret"
          
```

nit: new line

**Comment by btwseeu78** [comment]
@mbobrovskyi  you need add the lgtm label again , there was new line i just added a commit for it, sorry to bother and thanks for overall help and guide it means a lot.

**Comment by mimowo** [comment]
/retitle Helm: Parametrize setting IngressClass,priorityClass and imagePullSecret for KueueViz

File: charts/kueue/README.md [hunk 16]
```diff
@@ -126,9 +126,17 @@ The following table lists the configurable parameters of the kueue chart and the
...
+| kueueViz.backend.ingress.ingressClassName | string | `nil` | KueueViz dashboard backend ingress class name |
+| kueueViz.backend.ingress.tlsSecretName | string | `"kueueviz-backend-tls"` | KueueViz dashboard backend ingress tls secret name |
 | kueueViz.frontend.image.pullPolicy | string | `"Always"` | KueueViz dashboard frontend image pullPolicy. This should be set to 'IfNotPresent' for released version |
 | kueueViz.frontend.image.repository | string | `"us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueueviz-frontend"` | KueueViz dashboard frontend image repository |
 | kueueViz.frontend.image.tag | string | `"main"` | KueueViz dashboard frontend image tag |
+| kueueViz.frontend.ingress.host | string | `"frontend.kueueviz.local"` | KueueViz dashboard frontend ingress host |
+| kueueViz.frontend.ingress.ingressClassName | string | `nil` | KueueViz dashboard frontend ingress class name |
+| kueueViz.frontend.ingress.tlsSecretName | string | `"kueueviz-frontend-tls"` | KueueViz dashboard frontend ingress tls secret name |
+| kueueViz.imagePullSecrets | list | `[]` |  |
```
Comment (mimowo) [inline charts/kueue/README.md]: Please add description what this parameter is controlling

File: charts/kueue/README.md [hunk 17]
```diff
@@ -126,9 +126,17 @@ The following table lists the configurable parameters of the kueue chart and the
...
+| kueueViz.priorityClassName | string | `nil` |  |
```
Comment (mimowo) [inline charts/kueue/README.md]: Please add description what this parameter is controlling

**Comment by mimowo** [comment]
"4. Added test cases for kueueviz components in helm unit-test " should not be in release note.

/release-note-edit
```release-note
Helm: introduce new parameters to configure KueueViz installation:
- kueueViz.backend.ingress and kueueViz.frontend.ingress to configure ingress
- kueueViz.imagePullSecrets and kueueViz.priorityClassName
```

File: charts/kueue/values.yaml [hunk 18]
```diff
@@ -208,7 +208,16 @@ mutatingWebhook:
   # -- MutatingWebhookConfiguration's reinvocationPolicy
   reinvocationPolicy: Never
 kueueViz:
+  imagePullSecrets: []
```
Comment (mimowo) [inline charts/kueue/values.yaml]: Add comments so that the documentation is generated

File: charts/kueue/README.md (hunk 16 above)
Comment (mimowo) [inline charts/kueue/README.md]: Actually, I think it is better to move the parameters to per component frontend / backend. Yes, it means a bit of duplication, but it feels better to make it consistent and per Pod. wdyt?

Comment (btwseeu78) [reply charts/kueue/README.md]: apart from priorityClass and imagepullSecret its dedicated per componnet.

we can technically set diff priorityClass but kueueViz has one frontend and backend so any of this pending means the entire set is down. same for imagepullsecrets, i doubt that people will use 3 diff repo. i want to split the repo to ,registry ,repository and tags. registry can be set globally,but for later in other MR or dicussion we can decide.
i will fix the readme

Comment (mimowo) [reply charts/kueue/README.md]: Yeah, I know in typical case having two params means duplication, but technically they can be different, so I would suggest to have two params (one for frontend one for backend).

**Comment by btwseeu78** [comment]
@mimowo its not possible to rebase since latest addition to processing plan completely breaks the manifest generation add.
probably due to : [](https://github.com/kubernetes-sigs/kueue/pull/5820), the PR itself is not an issue but procession plan is definitely an issue, since its depends on hierarchy

**Comment by mimowo** [comment]
> @mimowo its not possible to rebase since latest addition to processing plan completely breaks the manifest generation add.

Yes, I merged today a PR touching the processing plan and so it probably conflicts.

Maybe you can re-apply the changes manually onto latest main?

**Comment by btwseeu78** [comment]
Same issue i checked, i will need to rebase and fix it taking both into account i will do it this weekend

File: Makefile [hunk 19]
```diff
@@ -187,6 +187,10 @@ helm-verify: helm helm-lint ## run helm template and detect any rendering failur
 # test kueueViz frontend nodeSelector and tolerations
 	$(HELM) template charts/kueue --set enableKueueViz=true --set kueueViz.frontend.nodeSelector.nodetype=infra --set 'kueueViz.frontend.tolerations[0].key=node-role.kubernetes.io/master' --set 'kueueViz.frontend.tolerations[0].operator=Exists' --set 'kueueViz.frontend.tolerations[0].effect=NoSchedule' > /dev/null
```
Comment (mbobrovskyi) [inline Makefile]: nit: remove line

File: Makefile [hunk 20]
```diff
@@ -187,6 +187,10 @@ helm-verify: helm helm-lint ## run helm template and detect any rendering failur
...
 
+# test kueueViz priorityClassName options for backend
+	$(HELM) template charts/kueue --set enableKueueViz=true --set kueueViz.backend.priorityClassName="system-cluster-critical" > /dev/null
+# test kueueViz priorityClassName options for frontend
+	$(HELM) template charts/kueue --set enableKueueViz=true --set kueueViz.backend.priorityClassName="system-cluster-critical" > /dev/null
```
Comment (mbobrovskyi) [inline Makefile:192]: nit: add new line

File: charts/kueue/values.yaml [hunk 21]
```diff
@@ -217,6 +217,19 @@ kueueViz:
     nodeSelector: {}
     # -- KueueViz backend tolerations
     tolerations: []
+    # -- Sets ImagePullSecrets for KueueViz dashboard backend and frontend deployments
```
Comment (mbobrovskyi) [inline charts/kueue/values.yaml]:
```suggestion
```
I think this no need.

File: charts/kueue/README.md [hunk 22]
```diff
@@ -128,12 +128,22 @@ The following table lists the configurable parameters of the kueue chart and the
 | kueueViz.backend.image.pullPolicy | string | `"Always"` | KueueViz dashboard backend image pullPolicy. This should be set to 'IfNotPresent' for released version |
 | kueueViz.backend.image.repository | string | `"us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueueviz-backend"` | KueueViz dashboard backend image repository |
 | kueueViz.backend.image.tag | string | `"main"` | KueueViz dashboard backend image tag |
+| kueueViz.backend.imagePullSecrets | list | `[]` | Sets ImagePullSecrets for KueueViz dashboard backend deployments This is useful when the images are in a private registry. |
```
Comment (mbobrovskyi) [inline charts/kueue/README.md]:
```suggestion
| kueueViz.backend.imagePullSecrets | list | `[]` | Sets ImagePullSecrets for KueueViz dashboard backend deployments. This is useful when the images are in a private registry. |
```
nit

File: charts/kueue/README.md [hunk 23]
```diff
@@ -128,12 +128,22 @@ The following table lists the configurable parameters of the kueue chart and the
...
+| kueueViz.backend.ingress.ingressClassName | string | `nil` | KueueViz dashboard backend ingress class name |
+| kueueViz.backend.ingress.tlsSecretName | string | `"kueueviz-backend-tls"` | KueueViz dashboard backend ingress tls secret name |
 | kueueViz.backend.nodeSelector | object | `{}` | KueueViz backend nodeSelector |
+| kueueViz.backend.priorityClassName | string | `nil` | Enable PriorityClass for KueueViz dashboard backend deployments |
 | kueueViz.backend.tolerations | list | `[]` | KueueViz backend tolerations |
 | kueueViz.frontend.image.pullPolicy | string | `"Always"` | KueueViz dashboard frontend image pullPolicy. This should be set to 'IfNotPresent' for released version |
 | kueueViz.frontend.image.repository | string | `"us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueueviz-frontend"` | KueueViz dashboard frontend image repository |
 | kueueViz.frontend.image.tag | string | `"main"` | KueueViz dashboard frontend image tag |
+| kueueViz.frontend.imagePullSecrets | list | `[]` | Sets ImagePullSecrets for KueueViz dashboard frontend deployments This is useful when the images are in a private registry. |
```
Comment (mbobrovskyi) [inline charts/kueue/README.md]:
```suggestion
| kueueViz.frontend.imagePullSecrets | list | `[]` | Sets ImagePullSecrets for KueueViz dashboard frontend deployments. This is useful when the images are in a private registry. |
```

File: charts/kueue/values.yaml [hunk 24]
```diff
@@ -230,6 +243,19 @@ kueueViz:
     nodeSelector: {}
     # -- KueueViz frontend tolerations
     tolerations: []
+    # -- Sets ImagePullSecrets for KueueViz dashboard backend and frontend deployments
```
Comment (mbobrovskyi) [inline charts/kueue/values.yaml]:
```suggestion
```

File: charts/kueue/tests/kueue_test.yaml [hunk 25]
```diff
@@ -0,0 +1,131 @@
...
+      kueueViz:
+        frontend:
+          imagePullSecrets:
+            - name: "my-secret"
+    asserts:
+      - equal:
+          path: spec.template.spec.imagePullSecrets[0].name
+          value: "my-secret"
+
+
```
Comment (mbobrovskyi) [inline charts/kueue/tests/kueue_test.yaml:131]: nit: new line

**Review by mbobrovskyi — COMMENTED** [review]
LGTM. Just few nits.

File: charts/kueue/tests/kueue_test.yaml (hunk 25 above)
Comment (mbobrovskyi) [inline charts/kueue/tests/kueue_test.yaml]:
```suggestion

```
nit

**Comment by mimowo** [comment]
@btwseeu78 please address the remaining comments and I'm happy to merge.

While cherrypick to 0.11 is out of question, would you like to cherrypick to 0.12?

**Comment by btwseeu78** [comment]
> @btwseeu78 please address the remaining comments and I'm happy to merge.
> 
> While cherrypick to 0.11 is out of question, would you like to cherrypick to 0.12?

yes im fine with .12

**Comment by mimowo** [comment]
/lgtm
/approve
Let's try:
/cherrypick release-0.12

**Comment by mimowo** [comment]
@btwseeu78  if you want feel free to prepare the cherrypick for 0.12. Otherwise I'm also find to just wait a bit for 0.13

**Comment by tenzen-y** [comment]
> @btwseeu78 if you want feel free to prepare the cherrypick for 0.12. Otherwise I'm also find to just wait a bit for 0.13

SGTM
